### PR TITLE
Upgrade to cats-effect 3.0.0 final and fs2 3.0.0 final

### PR DESCRIPTION
**Pull request by CremboC**
Supersedes https://github.com/http4s/http4s/pull/4662.
I ensured everything compiles locally via `sbt compile test:compile` but haven't run tests.

### CODE REVIEW

File: project/Http4sPlugin.scala [hunk 1]
```diff
@@ -34,9 +34,9 @@ object Http4sPlugin extends AutoPlugin {
   override lazy val buildSettings = Seq(
     // Many steps only run on one build. We distinguish the primary build from
     // secondary builds by the Travis build number.
-    http4sApiVersion := version.map {
-      case VersionNumber(Seq(major, minor, _*), _, _) => (major.toInt, minor.toInt)
-    }.value,
+    http4sApiVersion := version.map { case VersionNumber(Seq(major, minor, _*), _, _) =>
+      (major.toInt, minor.toInt)
+    }.value
```
Comment (CremboC) [inline project/Http4sPlugin.scala:39]: Hmm wasn't me that changed this, I did run `scalafmtCheck` though. Should probably revert these changes?

Comment (CremboC) [reply project/Http4sPlugin.scala:39]: Ah, I think I got `scalafmt` to run automatically and it probably was never run for this file, hence it was formatted when I touched it.

**Comment by CremboC** [comment]
There are some test failures, but I am mostly worried about:
```
java.lang.NoSuchMethodError: fs2.Pull.stream(Lscala/Predef$$less$colon$less;)Lfs2/Stream;
```

File: project/Http4sPlugin.scala (hunk 1 above)
Comment (rossabaker) [reply project/Http4sPlugin.scala:39]: We haven't been running scalafmt on sbt.  I'd like to, but we've been deferring that until the branch situation settles.

If you have a moment to revert that to keep this diff smaller, that would great.  If not, I can sort it out.

**Review by rossabaker — COMMENTED** [review]
Thanks for this!  This is a big help.  And I forgot about jawn-fs2 while I've been working on the others.  I'll get that lined up ASAP, and probably push it here so we can see the tests run.

File: project/Http4sPlugin.scala (hunk 1 above)
Comment (CremboC) [reply project/Http4sPlugin.scala]: I have reverted the changes and force pushed to keep clean history, hope that's ok 😬

**Comment by rossabaker** [comment]
:+1: once we green it up.  I have a PR open for jawn-fs2, and am ready to release vault, keypool, and log4cats.  That should line up the whole CE3 ecosystem.

**Comment by prova** [comment]
Was about to jump on it. Your jawn-fs2 PR has built fine.

**Comment by prova** [comment]
Thanks, no worries, fair play :-) I'm portig a fair bit of code now. CE3 and friends is a shot in the arm.

**Review by ChristopherDavenport — APPROVED** [review]
